src/context/product-context.js

```diff
@@ -0,0 +1,52 @@
+import { createContext,useContext, useEffect,useReducer  } from "react";
+import axios from "axios";
+import { sortReducer } from "../reducer/product-filter-reducer";
+import { sortPrice,filteredProduct,filterPrice,filterCategory,filterRating} from "../utilities/utils";
+
+
+
+const ProductContext = createContext();
+
+const initialState={
+    dataList: [],
+    sortBy: null,
+    setError: null,
+    showAllProduct: true,
+    priceValue: 0,
+    sortByCategory: [],
+    sortByRating: null
+    
+
+}
+
+const ProductProvider = ({children}) => {
+  const [state, dispatch] = useReducer(sortReducer, initialState);
+
```

Comment:
in future you would be having multiple context for ecomm so name your state and dispatch specific to the context like filterState and filterDispatch

Reply:
Done the suggested changes